src/TrackArtist.cpp

```diff
@@ -2114,22 +2113,23 @@ void TrackArtist::DrawClipSpectrum(WaveTrackCache &waveTrackCache,
    enum { DASH_LENGTH = 10 /* pixels */ };
 
    const ClipParameters params(true, track, clip, rect, selectedRegion, zoomInfo);
-   const wxRect &hiddenMid = params.hiddenMid;
+   const wxRect hiddenMid = params.hiddenMid;
+
    // The "hiddenMid" rect contains the part of the display actually
    // containing the waveform, as it appears without the fisheye.  If it's empty, we're done.
    if (hiddenMid.width <= 0) {
       return;
    }
 
-   const double &t0 = params.t0;
-   const double &tOffset = params.tOffset;
-   const double &ssel0 = params.ssel0;
-   const double &ssel1 = params.ssel1;
-   const double &averagePixelsPerSample = params.averagePixelsPerSample;
-   const double &rate = params.rate;
-   const double &hiddenLeftOffset = params.hiddenLeftOffset;
-   const double &leftOffset = params.leftOffset;
-   const wxRect &mid = params.mid;
+   const double t0 = params.t0;
+   const double tOffset = params.tOffset;
+   const double ssel0 = params.ssel0;
+   const double ssel1 = params.ssel1;
+   const double averagePixelsPerSample = params.averagePixelsPerSample;
+   const double rate = params.rate;
+   const double hiddenLeftOffset = params.hiddenLeftOffset;
+   const double leftOffset = params.leftOffset;
+   const wxRect mid = params.mid;
```

Comment:
Contra James, I still don't like this change.

params is effectively a bunch of stack variables, calculated in a constructor, which I abstracted out once to remove repetition of code between this function and the one that draw waves.

The declarations of references are a way to tell the compiler that these are abbreviated names for the members of params, when they are used later in the function.  That is not at all bizarre to me.  It's a way of doing DRY (don't repeat yourself).  I don't repeat an expression like params.t0, having reduced that to just a name.

I believe that the declaration of references implies no more growth of the local stack frame, whereas the removal of & causes more space to be allocated on the stack.

Prove me wrong:  show me disassemblies with and without the change.